lua/entities/gmod_tardis/modules/sh_tracking.lua

```diff
@@ -0,0 +1,497 @@
+                self.trackingdebugprop:Remove()
+            end
+        end
+
+        if IsValid(ply) then
+            if wasTracking ~= isTracking then
+                TARDIS:StatusMessage(ply, "Controls.Tracking.Status", IsValid(ent))
+            end
+            if isTracking then
+                if wasTrackingEnt ~= ent then
+                    local name = ent.PrintName or (isfunction(ent.Name) and ent:Name()) or ent.Name or ent:GetModel() or ent:GetClass()
+                    if ent.GetCreator then
+                        local creator = ent:GetCreator()
+                        if IsValid(creator) then
+                            name = name .. " (" .. creator:Nick() .. ")"
+                        end
+                    end
+                    TARDIS:Message(ply, "%s %s", "Controls.Tracking.Target", name)
+                else 
+                    TARDIS:Message(ply, "Controls.Tracking.SameTarget")
+                end
+            end
+        end
+        return true
+    end
+
+    function ENT:GetTrackRotation()
+        return self:GetData("tracking-rotation", false)
+    end
+
+    function ENT:SetTrackRotation(on)
+        self:SetData("tracking-rotation", on)
+
+        local ent = self:GetTracking()
+        if not IsValid(ent) then return end
+        local offsetPos = self:GetData("tracking-offset-pos", Vector(0,0,0))
+        local offsetYaw = self:GetData("tracking-offset-yaw", 0)
+        local newOffsetPos, newOffsetYaw
+        if on then
+            local currentTrackingPos = ent:GetPos() + offsetPos
+            newOffsetPos = ent:WorldToLocal(currentTrackingPos)
+            newOffsetYaw = offsetYaw - ent:GetAngles().y
+        else
+            local currentTrackingPos = ent:LocalToWorld(offsetPos)
+            newOffsetPos = WorldToLocal(currentTrackingPos, angle_zero, ent:GetPos(), angle_zero)
+            newOffsetYaw = offsetYaw + ent:GetAngles().y
+        end
+        self:SetData("tracking-offset-pos", newOffsetPos)
+        self:SetData("tracking-offset-yaw", newOffsetYaw)
+    end
+
+    function ENT:ToggleTrackRotation()
+        self:SetTrackRotation(not self:GetTrackRotation())
+    end
+
+    ENT:AddHook("Initialize", "tracking", function(self)
+        self:SetTrackRotation(true)
+    end)
+
+    ENT:AddHook("HandleE2", "tracking", function(self, name, e2, ...)
+        local args = {...}
+        if name == "Track" and (args[1] == e2.player or TARDIS:CheckPP(e2.player, self)) then
+            local ent = args[1]
+            local success = self:SetTracking(ent, e2.player)
+            return success and 1 or 0
+        end
+    end)
+
+    ENT:AddHook("HandleE2", "tracking_get", function(self, name, e2)
+        if name == "GetTracking" then
+            return self:GetTracking() or NULL
+        end
+    end)
+
+    ENT:AddHook("FlightToggled", "tracking", function(self, on)
+        if not on then
+            self:SetTracking(nil, self:GetData("pilot"))
+        end
+    end)
```

Comment:
Not particularly related to these lines
Should we perhaps do the same for physlock enable?
The distance condition will probably turn it off after a while, but do we need it to wait?

Reply:
https://github.com/MattJeanes/TARDIS/pull/969/commits/3cfba78974d82469dd97008a3b366cbb3605ddc0 + https://github.com/MattJeanes/TARDIS/pull/969/commits/e87c55689360020c41f91ba39b676e0f9925a2c0

Reply:
It also now auto disables physlock if it's engaged when activating tracking, like how it enables flight

https://github.com/MattJeanes/TARDIS/pull/969/commits/6bea59f4a5d583fbfd2f63d7553911e575638ea3

Reply:
The first thing is great
I am not sure about "auto-disabling physlock" part

Seems like making things too easy
If you want that, perhaps it could be a setting? 

Same situation as auto-close doors during teleport, I believe it takes away a part of gameplay
The TARDIS as we've always known it is supposed to make you press a bunch of levers and buttons, run around the console etc - not automate controls for you

Reply:
Happy to remove, just thought might make things easier - but do we want it to fail to engage tracking if physlock is enabled or let it enable but just not move until physlock is disabled?

Reply:
I'm happy with it just failing
But that's not a strong preference

Reply:
https://github.com/MattJeanes/TARDIS/pull/969/commits/65525f55a4954fd62d934d9f9d782735378f162c